torch/_inductor/select_algorithm.py

```diff
@@ -1338,7 +1338,8 @@ def wait_on_futures():
             return wait_on_futures
 
         def autotune(choices):
-            return make_benchmark_fn()(choices)
+            with dynamo_timed("template_autotuning"):
+                return make_benchmark_fn()(choices)
```

Comment:
Have you spot checked this ? does it cover the benchmarking and the compilation ? i'm not exactly sure how we should record the precompilation, since it is async..

Reply:
I think we also want the time in `benchmark_fused_nodes`

Reply:
I created a toy script to do matmul and spot check the result as:
```
(Pdb) [U.compilation_time_metrics[key] for key in ["pad_mm_benchmark", "template_autotuning"]]
[[4.7206878662109375e-05], [2.0357143878936768]]
```

This should only cover the autotuning. Precompilation is parallel but we do wait for all `futures` to be fulfilled at some point. We can probably log that wait time

Reply:
I have no calibration for whether or not these figures are correct or not. Do these match your expectations? In particular pad_mm_benchmark seems really fast!

Reply:
pad_mm_benchmark will be fast if there are no unaligned mms to benchmark.

Reply:
I think it's expected since the tensor sizes I pick are quite regular. Some heuristics probably skips benchmarking for padding and make it super fast

Reply:
I think it's due to this check in mm padding: https://github.com/pytorch/pytorch/blob/ff616c26fb66b26966b6f6f50aa85bf615400816/torch/_inductor/fx_passes/pad_mm.py#L413 .  cc @ezyang